app/components/templates/CreateTemplateModal/useFormState.ts

```diff
@@ -6,46 +6,27 @@ interface FormState {
     tenants: string[];
     pages: string[];
 }
-
-const useFormState = (
-    initialState: Partial<FormState>
-): [FormState, (value: Partial<FormState>) => void, () => void] => {
-    const [templateName, setTemplateName] = useState<string>(initialState.templateName || '');
-    const [roles, setRoles] = useState<string[]>(initialState.roles || []);
-    const [tenants, setTenants] = useState<string[]>(initialState.tenants || []);
-    const [pages, setPages] = useState<string[]>(initialState.pages || []);
+const defaultState: FormState = {
+    templateName: '',
+    roles: [],
+    tenants: [],
+    pages: []
+};
+const useFormState = (initialState: Partial<FormState>) => {
+    const [formState, setFormState] = useState<FormState>({ ...defaultState, ...initialState });
 
     const setState = useCallback(
-        (value: Partial<FormState>) => {
-            Object.entries(value).forEach(([key, val]) => {
-                switch (key) {
-                    case 'templateName':
-                        setTemplateName(val as string);
-                        break;
-                    case 'roles':
-                        setRoles(val as string[]);
-                        break;
-                    case 'tenants':
-                        setTenants(val as string[]);
-                        break;
-                    case 'pages':
-                        setPages(val as string[]);
-                        break;
-                    default:
-                }
-            });
+        (newState: Partial<FormState>) => {
+            setFormState(prevState => ({ ...prevState, ...newState }));
         },
-        [setTemplateName, setRoles, setTenants, setPages]
+        [formState, setFormState]
     );
 
     const clearState = useCallback(() => {
-        setTemplateName('');
-        setRoles([]);
-        setTenants([]);
-        setPages([]);
-    }, [setTemplateName, setRoles, setTenants, setPages]);
+        setFormState({ ...defaultState, ...initialState });
+    }, [formState, setFormState]);
```

Comment:
Shouldn't it be

```suggestion
    }, [initialState, setFormState]);
```

?